ludwig/schema/trainer.py

```diff
@@ -25,7 +26,7 @@ def wrap(trainer_config: BaseTrainerConfig):
 
 
 @dataclass
-class BaseTrainerConfig(schema_utils.BaseMarshmallowConfig):
+class BaseTrainerConfig(schema_utils.BaseMarshmallowConfig, metaclass=ABCMeta):
```

Comment:
Why not just inherit from ABC?

```suggestion
class BaseTrainerConfig(schema_utils.BaseMarshmallowConfig, ABC):
```

https://docs.python.org/3/library/abc.html#abc.ABC

Reply:
good point!